BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -154,30 +200,116 @@ extension ElementContent.Builder: ContentStorage {
         return result
     }
 
-    private var layoutItems: [(LayoutType.Traits, Measurable)] {
-        return children.map { ($0.traits, $0) }
+    private func layoutItems(in environment: Environment) -> [(LayoutType.Traits, Measurable)] {
+        return children.map { ($0.traits, $0.content.measurable(in: environment)) }
     }
 
 }
 
 
 extension ElementContent.Builder {
 
-    fileprivate struct Child: Measurable {
+    fileprivate struct Child {
 
         var traits: LayoutType.Traits
         var key: String?
         var content: ElementContent
         var element: Element
 
-        func measure(in constraint: SizeConstraint) -> CGSize {
-            return content.measure(in: constraint)
-        }
+    }
+
+}
 
+private struct EnvironmentAdaptingStorage: ContentStorage {
+    let childCount = 1
+
+    var layout: SingleChildLayout
+
+    /// During measurement or layout, the environment adapter will be applied
+    /// to the environment before passing it
+    ///
+    var adapter: (inout Environment) -> Void
+
+    var child: Element
+
+    func performLayout(
+        attributes: LayoutAttributes,
+        environment: Environment)
+        -> [(identifier: ElementIdentifier, node: LayoutResultNode)]
+    {
+        let environment = adapted(environment: environment)
+
+        let childAttributes = layout.layout(
+            size: attributes.bounds.size,
+            child: child.content.measurable(in: environment))
+
+        let identifier = ElementIdentifier(key: nil, index: 0)
```

Comment:
Still not super familiar with identifiers, but are you sure there's no repercussions of this always being nil/0?

Reply:
This is fine because there's only ever one child, I believe.

Reply:
Yep, it's scoped to this element's single child, so literally any value would be fine.